x/observer/migrations/v9/migrate.go

```diff
@@ -0,0 +1,38 @@
+package v9
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/zeta-chain/node/x/observer/types"
+)
+
+type observerKeeper interface {
+	GetBallotListForHeight(ctx sdk.Context, height int64) (val types.BallotListForHeight, found bool)
+	DeleteBallot(ctx sdk.Context, index string)
+	DeleteBallotListForHeight(ctx sdk.Context, height int64)
+}
+
+const MaturityBlocks = int64(100)
+
+// MigrateStore migrates the x/observer module state from the consensus version 8 to version 9.
+// It updates the indexing for chain nonces object to use chain ID instead of chain name
+func MigrateStore(ctx sdk.Context, observerKeeper observerKeeper) error {
+	currentHeight := ctx.BlockHeight()
+	// Maturity blocks is a parameter in the emissions module
+	if currentHeight < MaturityBlocks {
+		return nil
+	}
+	maturedHeight := currentHeight - MaturityBlocks
+	for i := maturedHeight; i > 0; i-- {
+		ballotList, found := observerKeeper.GetBallotListForHeight(ctx, i)
+		if !found {
+			continue
+		}
+		for _, ballotIndex := range ballotList.BallotsIndexList {
+			observerKeeper.DeleteBallot(ctx, ballotIndex)
+		}
+		observerKeeper.DeleteBallotListForHeight(ctx, i)
+	}
```

Comment:
I seem to remember you saying this would take a long time right?

Reply:
yeah 
https://zetachain.slack.com/archives/C04GM8RV5E3/p1729172612290599

Reply:
> its taking areound 12 secs on mainnet and 21 secs on testnet data to run the delete.

yeah I think that's fine

Reply:
We can try to spin up 4-8 goroutines for this migration and test whether will be faster

Reply:
I actually tried with some goroutines earlier , something like this 
```
func MigrateStore(ctx sdk.Context, observerKeeper observerKeeper) error {
	currentHeight := ctx.BlockHeight()
	if currentHeight < MaturityBlocks {
		return nil
	}
	maturedHeight := currentHeight - MaturityBlocks

	var wg sync.WaitGroup
	numGoroutines := 10
	blockRange := maturedHeight / int64(numGoroutines)

	for i := 0; i < numGoroutines; i++ {
		start := int64(i) * blockRange
		end := start + blockRange
		if i == numGoroutines-1 {
			end = maturedHeight + 1
		}

		wg.Add(1)
		go func(start, end int64) {
			defer wg.Done()
			for j := start; j < end; j++ {
				ballotList, found := observerKeeper.GetBallotListForHeight(ctx, j)
				if !found {
					continue
				}
				for _, ballotIndex := range ballotList.BallotsIndexList {
					observerKeeper.DeleteBallot(ctx, ballotIndex)
				}
				observerKeeper.DeleteBallotListForHeight(ctx, j)
			}
		}(start, end)
	}

	wg.Wait()
	return nil
}
```

although it does show an improvement , 
Without goroutine is 11.5 secs 
With goroutine is 8.6 secs 

I decided to go in favour of the simpler logic as the difference in time was not as much .